graphql_test.go

```diff
@@ -74,6 +74,30 @@ func (v *customDirectiveVisitor) After(ctx context.Context, directive *types.Dir
 	return fmt.Sprintf("Directive '%s' modified result: %s", directive.Name.Name, output.(string)), nil
 }
 
+type cachedDirectiveVisitor struct {
+	cachedValue interface{}
+}
+
+func (v *cachedDirectiveVisitor) Before(ctx context.Context, directive *types.Directive, input interface{}) (bool, error) {
```

Comment:
Here I feel that we need some info about the field or its parent ID so that we can get the value from cache based on the field path/args or its parent ID.

Reply:
This can happen later when we implement selected fields.